doc/en/proposals/importmode_root_spec.rst

```diff
@@ -0,0 +1,71 @@
+:orphan:
+
+===================================
+PROPOSAL: Parametrize with fixtures
+===================================
+
+.. warning::
+
+    This document outlines a proposal around creating a new import mode that supports pep420 and
+    supports the dissonance between installed and editable installed
+
+
+Problem
+=======
+
+test module discovery in pytest currently is is pre-pep40 and create pains for users
+by adding anything things to sys.path and creating easy conflicts in sys.modules
+
+Additionally this will not solve the dissonance between doctest tests vs installed modules
+
+The importlib mode in contrast breaks all kinds of expectations by importing test modules in a magical way
+that breaks relative imports and leaves them out of ``sys.modules``
+
+
+Proposed Solution
+=================
+
+A new import mode take a definition of import roots.
+
+a import root is either a folder in the worktree that will no be installed,
+or a folder that will be installed either normally or editable.
```

Comment:
How is installation done? Like pytest will install to the venv with pip?

Reply:
Users Will have to manage install, pytest should never do that part